02-microarray/gene-id-annotation_microarray_01_ensembl.Rmd

```diff
@@ -297,47 +298,27 @@ head(multi_mapped)
 
 Looks like we have some cases where 3 gene symbols mapped to a single Ensembl ID.
 We have a total of 130 out of 17984 Ensembl IDs with multiple mappings to gene symbols.
-We are not too worried about the 130 IDs with multiple mappings, so we will filter them out for the purpose of having 1:1 mappings for our downstream analysis.
+If we are not too worried about the 130 IDs with multiple mappings, we can filter them out for the purpose of having 1:1 mappings for our downstream analysis.
 
 ## Map Ensembl IDs to gene symbols -- filtering out multi mappings
 
-The next code chunk will rerun the `mapIds()` function, this time supplying the `"filter"` option to the `multiVals` argument.
+The next code we chunk we will rerun the `mapIds()` function, this time supplying the `"filter"` option to the `multiVals` argument.
 This will remove all instances of multiple mappings and return a list of only the gene identifiers and symbols that had 1:1 mapping.
 Use `?mapIds` to see more options or strategies.
 
 ```{r}
 # Map ensembl IDs to their associated gene symbols
-filtered_mapped_df <- mapIds(
-  org.Mm.eg.db, # Replace with annotation package for the organism relevant to your data
-  keys = df$Gene,
-  column = "SYMBOL", # Replace with the type of gene identifiers you would like to map to
-  keytype = "ENSEMBL", # Replace with the type of gene identifiers in your data
-  multiVals = "filter"
-) %>%
-  # Turn our list into a data frame
-  reshape2::melt() %>%
-  # Now let's give the `value` column a name that is more relevant to the values it holds
-  dplyr::rename("Symbol" = "value")
-```
-
-Let's copy the same code we used above to see if we have any multi mappings in our new data frame.
-
-```{r}
-filtered_mapped_df %>%
-  # First, we need to grab the Ensembl IDs from the rownames
-  tibble::rownames_to_column("Gene") %>%
-  # Let's group by the Ensembl ID's in the `L1` column
-  dplyr::group_by(Gene) %>%
-  # Create a new variable containing the number of symbols mapped to each ID
-  dplyr::mutate(gene_symbol_count = dplyr::n()) %>%
-  # Arrange by the genes with the highest number of symbols mapped
-  dplyr::arrange(desc(gene_symbol_count)) %>%
-  # Filter to include only the rows with multi mappings
-  dplyr::filter(gene_symbol_count > 1)
+filtered_mapped_df <- data.frame(
```

Comment:
```suggestion
filtered_mapped_df <- data.frame(
     df, 
```
This isn't formatted properly, but you get the idea.

Reply:
@cansavvy, I get an error here due to the difference in length between `df` (nrows = 17918) and the results in `filtered_mapped_df` (nrows = 17854). Do you think we should pull the genes out of the rownames and do an `inner_join` here?

Reply:
Oh of course. Forgot about that. Will need an inner_join. I can take a look after add that if you want.

Reply:
Okay @cansavvy, you can take a look at the `inner_join` step and let me know if I need to make any changes 👍